static/redesign/apps/style/views/left/select-map-view.js

```diff
@@ -97,6 +96,7 @@ define(["jquery",
 
                 dataSources.forEach(function(dataSource) {
                     var collection = that.app.dataManager.getCollection(dataSource.value);
+                    if (collection.length < 1) {return;}
                         if (dataSource.value === "markers") {
                             var layer = new Layer({
                                 map_id: that.map.id,
```

Comment:
Nice work. Two brief comments:
* Whenever you get a chance, please rename the "class" key (line 184) to something other than "class" (which is a reserved word). Not a big deal -- it works, but just as a general rule.
* To finish the job of making this class as modular as possible, you may want to re-visit the dataSources.forEach loop, which has some repeating code across the two if conditions, and is kinda long (and hence a little hard to read). You may start by (a) making 2 functions, createLayerMarkers, and createLayerRecords, that can be called from within the loop, then (b) comparing the 2 functions to see what's similar and different, and finally (c) just making one function that includes an additional argument that enables you to add the customizations that are unique to the datatype.

No rush on this -- just a suggestion.